app/src/main/kotlin/com/saltedge/authenticator/features/consents/ConsentsListFragment.kt

```diff
@@ -0,0 +1,149 @@
+import com.saltedge.authenticator.tools.authenticatorApp
+import com.saltedge.authenticator.tools.loadRoundedImage
+import com.saltedge.authenticator.tools.stopRefresh
+import com.saltedge.authenticator.widget.fragment.BaseFragment
+import com.saltedge.authenticator.widget.list.SpaceItemDecoration
+import kotlinx.android.synthetic.main.fragment_consents_list.*
+import kotlinx.android.synthetic.main.fragment_consents_list.subTitleView
+import kotlinx.android.synthetic.main.fragment_consents_list.titleView
+import javax.inject.Inject
+
+const val KEY_CONSENTS = "consents"
+
+class ConsentsListFragment : BaseFragment(),
+    ListItemClickListener {
+
+    @Inject lateinit var viewModelFactory: ViewModelsFactory
+    private lateinit var viewModel: ConsentsListViewModel
+    private lateinit var binding: ConsentsListBinding
+    private val adapter = ConsentsListAdapter(clickListener = this)
+    private var headerDecorator: SpaceItemDecoration? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        authenticatorApp?.appComponent?.inject(this)
+        setupViewModel()
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        activityComponents?.updateAppbar(
+            titleResId = R.string.consents_feature_title,
+            backActionImageResId = R.drawable.ic_appbar_action_back
+        )
+        binding = DataBindingUtil.inflate(
+            inflater,
+            R.layout.fragment_consents_list,
+            container,
+            false
+        )
+        binding.viewModel = viewModel
+        binding.lifecycleOwner = this
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setupViews()
+    }
+
+    override fun onListItemClick(itemIndex: Int, itemCode: String, itemViewId: Int) {
+        viewModel.onListItemClick(itemIndex)
+    }
+
+    private fun setupViewModel() {
+        viewModel = ViewModelProvider(this, viewModelFactory)
+            .get(ConsentsListViewModel::class.java)
+        lifecycle.addObserver(viewModel)
+
+        viewModel.listItems.observe(this, Observer<List<ConsentItemViewModel>> {
+            headerDecorator?.setHeaderForAllItems(it.count())
+            headerDecorator?.footerPositions = arrayOf(it.count() - 1)
+            adapter.data = it
+        })
+        viewModel.connectionItem.observe(this, Observer<ConnectionViewModel> {
+            titleView?.text = it.name
+            subTitleView?.text = it.consentDescription
+            logoImageView?.loadRoundedImage(
+                imageUrl = it.logoUrl,
+                placeholderId = R.drawable.shape_bg_app_logo,
+                cornerRadius = resources.getDimension(R.dimen.connections_list_logo_radius)
```

Comment:
by design icon radius is not equal to `connections_list_logo_radius`, i equal to `6dp`. Discuss with Product Manager.